README.rst

```diff
@@ -15,6 +15,8 @@ the network).
 
 Installation
 ------------
+Argcomplete requires Python 2.7 or 3.3+.
```

Comment:
I moved it under its own heading, but I don't know if it looks weird because the section is so short.

I also removed another bit of 2.6-specific code and updated the RTD link.